Newsletters: add 399 (2026-04-03) (#2693)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2026-04-03-newsletter.md

---
title: 'Bitcoin Optech Newsletter #399'
permalink: /en/newsletters/2026/04/03/
name: 2026-04-03-newsletter
slug: 2026-04-03-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter describes how wallet fingerprinting can damage
payjoin privacy and summarizes a proposal for a wallet metadata backup
format. Also included are our regular sections summarizing proposals
and discussion about changing Bitcoin's consensus rules, announcing new
releases and release candidates, and describing notable changes to
popular Bitcoin infrastructure software.

## News

> - **How Wallet Fingerprinting Damages Payjoin Privacy**: This week Armin Sabouri

Author: Maybe?

```suggestion
- **Wallet fingerprinting risks for payjoin privacy**: This week Armin Sabouri
```

  [posted][topic payjoin fingerprinting] to Delving about how differences in how
  Payjoin implementations make it possible to fingerprint Payjoin transactions

>   and damages Payjoin's privacy.

Reviewer: Thanks, Fixed in [cb2290a](https://github.com/bitcoinops/bitcoinops.github.io/pull/2693/commits/cb2290a8e56f465181c6f528259ff0661368ebbe)

  Sabouri states that Payjoin transactions should appear indistinguishable from
  standard single-party transactions. With this fact, analyst who are aware of
  [Payjoin][topic payjoin] should first detect artifacts of collaborative transactions. This can
  be done at two levels, intra-transaction and inter-transaction.

  He then went on to take a look at 3 implementations of Payjoin, Samourai, and PDK
  Demo and Cake Wallet -> Bull Bitcoin Mobile. In each of these example he finds

>   a few discrepancies which make it possible to fingerprint these

Reviewer: I added what specific example he found in the post.

[cb2290a](https://github.com/bitcoinops/bitcoinops.github.io/pull/2693/commits/cb2290a8e56f465181c6f528259ff0661368ebbe)

  implementations. Overall, Sabouri makes the conclusion that while some of these
  wallet fingerprints are trivial to eliminate, and others are intrinsic to a
  particular wallet's design choice. Wallet developers should be aware of these
  potential privacy leaks when implementing Payjoin into their wallets.

> - **New BIP for a wallet backup metadata format**: Pythcoiner

Reviewer 2: Its a scheme for backup of wallet metadata

  public keys. Conduition refers to this as "rerandomization" and it
  fundamentally enables [BIP32][], [BIP341][], and [BIP352][] (with some
  additional cryptographic innovation, probably).

  To date, there are not signature aggregation protocols for IBC like there
  are in [MuSig][topic musig] and [FROST][topic threshold signature], and
  conduition issues a call to action for Bitcoin developers and cryptographers
  to research what may be possible.

  Keys and signatures in known IBC cryptosystems are about twice the
  size of keys in ECDLP-dependent cryptosystems. Much smaller than hash-based
  or lattice-based cryptosystems. Verification is costly even on desktop
  machines (on the order of 1 millisecond per verification), in the same
  ballpark as hash-based and lattice-based.

- **Varops budget and tapscript leaf 0xc2 (aka "Script Restoration") are BIPs 440 and 441**:
  Rusty Russel [wrote][rr ml gsr bips] on the Bitcoin Development Mailing List
  that the first two BIPs of the Great Script Restoration (or Grand Script
  Renaissance) have been submitted for BIP numbering. They subsequently
  received BIP numbers 440 and 441 respectively. [BIP440][news374 varops]

>   enables the restoration of Script by building an accounting system for the

Author: Maybe

```suggestion
  enables the restoration of previously-disabled Script opcodes by building an accounting system for the
```
